designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/AdditionalUIConfigFinalizer.scala

```diff
@@ -0,0 +1,69 @@
+package pl.touk.nussknacker.ui.definition
+
+import cats.implicits.catsSyntaxSemigroup
+import pl.touk.nussknacker.engine.api.component.{
+  AdditionalUIConfigProvider,
+  ComponentId,
+  ScenarioPropertyConfig,
+  SingleComponentConfig
+}
+import pl.touk.nussknacker.engine.api.process.ProcessingType
+import pl.touk.nussknacker.engine.definition.component.ComponentStaticDefinition
+import pl.touk.nussknacker.engine.definition.model.ModelDefinition
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+import pl.touk.nussknacker.ui.component.{ComponentAdditionalConfigConverter, DefaultComponentIdProvider}
+
+class AdditionalUIConfigFinalizer(
```

Comment:
this class handles also base components, meaning that the first TODO in `AdditionalUIConfigProvider` can be removed 
` * TODO: The current implementation allows providing configs only for standard components - meaning that base components aren't handled.`